include/sparrow/layout/layout_utils.hpp

```diff
@@ -73,4 +76,29 @@ namespace sparrow::detail
         layout_type* p_layout;
     };
 
-}; // namespace sparrow::detail
+} // namespace sparrow::detail
+
+
+template<class OFFSET_TYPE>
+concept offset_type = std::is_same<std::remove_const_t<OFFSET_TYPE>, std::uint32_t>::value || 
+                        std::is_same<std::remove_const_t<OFFSET_TYPE>, std::uint64_t>::value;
+
+template<offset_type OFFSET_TYPE, std::ranges::range SIZES_RANGE>
+requires(std::unsigned_integral<std::ranges::range_value_t<SIZES_RANGE>>)
+u8_buffer<OFFSET_TYPE> offset_buffer_from_sizes(SIZES_RANGE && sizes)
```

Comment:
Is that specific to the different list layouts? If so, it should be moved to `list_array.hpp`. In any case I think it can be put in the `detail` namespace.

Reply:
other layouts need offsets from sizes I believe